lib/combinePrimitives.js

```diff
@@ -1,255 +1,252 @@
 'use strict';
 var Cesium = require('cesium');
-var defined = Cesium.defined;
-var defaultValue = Cesium.defaultValue;
-var DeveloperError = Cesium.DeveloperError;
+var deepEqual = require('deep-equal');
+var PrimitiveHelpers = require('./PrimitiveHelpers');
+var createAccessor = require('./createAccessor');
+var readAccessor = require('./readAccessor');
 
-var byteLengthForComponentType = require('./byteLengthForComponentType');
-var getAccessorByteStride = require('./getAccessorByteStride');
-var numberOfComponentsForType = require('./numberOfComponentsForType');
-var readBufferComponent = require('./readBufferComponent');
-var writeBufferComponent = require('./writeBufferComponent');
+var WebGLConstants = Cesium.WebGLConstants;
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
 
 module.exports = combinePrimitives;
 
-//Combines primitives in a mesh which have the same material, mode, and attributes.
+/**
+ * Combines all of the provided primitives if possible.
+ * If primitives have no shared data with the other primitives, their attributes and indices will be concatenated together and
+ * the indices will be incremented by the attribute offset. If the primitives have the same attribute accessors but different index
+ * accessors, the index accessors will be concatenated. In all other cases, the primitives will be left alone.
+
+ * @param {Object} gltf A javascript object containing a glTF asset.
+ * @returns {Object} The glTF asset with combined primitives.
+ *
+ * @see addPipelineExtras
+ * @see loadGltfUris
+ * @see combineMeshes
+ */
 function combinePrimitives(gltf) {
-    if (defined(gltf.accessors) && defined(gltf.bufferViews) && defined(gltf.buffers)) {
-        var meshes = gltf.meshes;
-        if (defined(meshes)) {
-            for (var meshId in meshes) {
-                if (meshes.hasOwnProperty(meshId)) {
-                    var mesh = meshes[meshId];
-                    var primitives = mesh.primitives;
-                    if (defined(primitives)) {
-                        mesh.primitives = combineMeshPrimitives(gltf, meshId, primitives);
+    PrimitiveHelpers.markPrimitiveConflicts(gltf, PrimitiveHelpers.getAllPrimitives(gltf));

```

Comment:
It's a little awkward to save the conflicts in the pipeline object. If possible try to keep it in here.